### PR TITLE
Fix issue #1575 distinguish between 'cased' and 'non-cased' characters

### PR DESCRIPTION
**Pull request by Regexp96**
#1575 distinguish between 'cased' and 'non-cased' characters

use unic::ucd::is_cased function to ignore non-cased character

### CODE REVIEW

**Comment by coolreader18** [comment]
This looks good! Could you add some tests in `tests/snippets/strings.py` making sure that the new behavior is correct?

**Comment by Regexp96** [comment]
ok I add some tests
thanks for reviewing

File: vm/src/obj/objstr.rs [hunk 1]
```diff
@@ -725,25 +726,32 @@ impl PyString {
...
     #[pymethod]
     fn isupper(&self, _vm: &VirtualMachine) -> bool {
-        // TODO: assert not " ".isupper(), assert not "_".isupper()
-        !self.value.is_empty()
-            && self
-                .value
-                .chars()
-                .filter(|x| !x.is_ascii_whitespace() && *x != '_')
-                .all(char::is_uppercase)
+        let mut cased = false;
```
Comment (youknowone) [inline vm/src/obj/objstr.rs]: how about this way?
```rust
   let mut cased = true;
   for c in self.value.chars() {
       if is_cased(c) && c.is_uppercase() {
            continue
        } else {
           cased = false;
           break;  // or `return true` as original
       }
   }
   cased
```

Comment (Regexp96) [reply vm/src/obj/objstr.rs]: oh it's looks better!

Comment (Regexp96) [reply vm/src/obj/objstr.rs]: this code return `false` when `"111a".islower()`
but in python it will return `true`

**Pull request merged**